### PR TITLE
feat: Backwards-Compatible Auth

### LINKED ISSUE #1402: `GoogleAuth` can no longer be used with `googleapis`: Type 'GoogleAuth<JSONClient>' is not assignable to type 'string | OAuth2Client | BaseExternalAccountClient | GoogleAuth<JSONClient> | undefined'
After updating `google-auth-library-nodejs` from v7.14.1 to v8.0.1, the following Typescript compile error occurs:
```
Type 'GoogleAuth<JSONClient>' is not assignable to type 'string | OAuth2Client | BaseExternalAccountClient | GoogleAuth<JSONClient> | undefined'.
  Type 'import(".../node_modules/google-auth-library/build/src/auth/googleauth").GoogleAuth<import(".../node_modules/google-auth-library/build/src/auth/googleauth").JSONClient>' is not assignable to type 'import(".../node_modules/google-gax/node_modules/google-auth-library/build/src/auth/googleauth").GoogleAuth<import(".../node_modules/google-gax/node_modules/google-auth-library/build/src/auth/googleauth").JSONClient>'.
    Types have separate declarations of a private property 'checkIsGCE'.
```

#### Environment details

  - OS: Windows
  - Node.js version: v16.14.2
  - npm version: v8.7.0
  - `google-auth-library` version: v8.0.1

#### Steps to reproduce

  1. Use the same code as previously
```typescript
import { google } from 'googleapis'
import { GoogleAuth } from 'google-auth-library'

function setCredentials(): void {
  const client = new GoogleAuth({
    scopes: ['https://www.googleapis.com/auth/cloud-billing', 'https://www.googleapis.com/auth/cloud-platform'],
  })

  google.options({
    auth: client, // Error is emmitted for this line after updating google-auth-library-nodejs to v8
  })
}
```
  3. Execute `tsc` for Typescript compile

### PR DESCRIPTION
**Pull request by danielbankhead**
This will greatly improve the experience for customers transitioning between different versions of `google-auth-library`.

Consumers can use the new `GoogleAuth.normalize` and `AuthClient.normalize` APIs to ensure the provided `GoogleAuth` and `AuthClient` instances are compatible with the current `google-auth-library` version.

Example:

```ts
// say this depends on an older version of `google-auth-library` (e.g. `v8.9.0`)
import { google } from 'googleapis';

// say this is the latest version of `google-auth-library` (e.g. `v9.0.0`)
import { GoogleAuth } from 'google-auth-library';

const auth = new GoogleAuth();

// this used to fail, but should work after internally using `instanceof GoogleAuth`.
google.options({auth});
```

Fixes #1402 🦕

### CODE REVIEW

**Comment by danielbankhead** [comment]
`do not merge` added until internal proposal has been approved.

**Comment by danielbankhead** [comment]
Another issue that would be resolved with this proposal: https://github.com/googleapis/google-api-nodejs-client/issues/3357

File: test/fixtures/key.p12
Comment (ruyadorno) [inline test/fixtures/key.p12:1]: is this related to the changes here?

File: test/test.jwt.ts
Comment (ruyadorno) [inline test/test.jwt.ts:31]: same here, was it just an unused value?

Comment (danielbankhead) [reply test/test.jwt.ts:31]: Yep, an unused value

File: test/fixtures/key.p12
Comment (danielbankhead) [reply test/fixtures/key.p12:1]: not really; noticed it was an unused file that we can remove

**Review by ruyadorno — APPROVED** [review]
LGTM 👍

**Comment by danielbankhead** [comment]
This would introduce more complexity than is required to solve this problem; instead, we should encourage dependencies to expose their version of `google-auth-library` so that customers can have a guaranteed, consistently working version.

**Pull request closed without merging**
